tools/scripts/rebuildAnimationLibraryManifest.rb

```diff
@@ -0,0 +1,217 @@
+      metadata['sourceUrl'] = "/api/v1/animation-library/#{metadata['version']}/#{name}.png"
+
+      # Populate sourceSize if not already present
+      unless metadata.key?('sourceSize')
+        png_body = objects['png'].object.get.body.read
+        metadata['sourceSize'] = dimensions_from_png(png_body)
+      end
+
+      animation_metadata_by_name[name] = metadata
+      verbose <<-EOS
+#{bold name} @ #{metadata['version']}
+#{JSON.pretty_generate metadata}
+      EOS
+    end
+
+    info "Metadata built for #{animation_metadata_by_name.size} animations."
+
+    info "Building alias map..."
+
+    alias_map = {}
+    animation_metadata_by_name.each do |name, metadata|
+      aliases = [name] + (metadata['aliases'] || [])
+      aliases.each do |aliaz|
+        # Push name into target array, deduplicate, and sort
+        alias_map[aliaz] = ((alias_map[aliaz] || []) + [name]).uniq.sort
+      end
+    end
+
+    alias_map.each {|k, v| verbose "#{bold k}: #{v.join(', ')}"} if @options[:verbose]
+
+    info "Mapped #{alias_map.size} aliases."
+
+    # Write result to file
+    File.open(DEFAULT_OUTPUT_FILE, 'w') do |file|
+      file.write(JSON.pretty_generate({
+          'metadata': animation_metadata_by_name,
+          'aliases': alias_map
+      }))
+    end
+    info <<-EOS.unindent
+      Manifest written to #{DEFAULT_OUTPUT_FILE}.
+
+        #{dim 'd[ o_0 ]b'}
+    EOS
+
+  # Report any issues while talking to S3 and suggest most likely steps for fixing it.
+  rescue Aws::Errors::ServiceError => service_error
+    warn service_error.inspect
+    warn <<-EOS.unindent
+
+      #{bold 'There was an error talking to S3.'}  Make sure you have credentials set using one of:
+
+        * aws_access_key and aws_secret_key in your locals.yml
+        * ENV['AWS_ACCESS_KEY_ID'] and ENV['AWS_SECRET_ACCESS_KEY']
+        * ~/.aws/credentials
+
+      #{dim 'See http://docs.aws.amazon.com/sdkforruby/api/Aws/S3/Client.html for more details.'}
+    EOS
+  end
+
+  def verbose(s)
+    puts(s) if @options[:verbose]
+  end
+
+  def info(s)
+    puts(s) unless @options[:quiet]
+  end
+
+  def warn(s)
+    puts(s)
+  end
+end
+
+# Given the complete body of a PNG file, returns hash of source dimensions:
+# {"x": _, "y": _}
+def dimensions_from_png(png_body)
+  # Read the first eight bytes of the IHDR Chunk, which must always be the
+  # first chunk of the PNG file.
+  #
+  # PNG Header takes 8 bytes (0x00-0x07)
+  # IHDR chunk length takes 4 bytes (0x08-0x0b)
+  # IHDR chunk type code takes 4 bytes (0x0c-0x0f)
+  # IHDR chunk data begins at 0x10
+  #
+  # The IHDR chunk begins with
+  # 4 bytes for width (0x10-0x13) followed by
+  # 4 bytes for height (0x14-0x17)
+  #
+  # PNG File Structure
+  # http://www.libpng.org/pub/png/spec/1.2/PNG-Structure.html
+  # IHDR Chunk Layout
+  # http://www.libpng.org/pub/png/spec/1.2/PNG-Chunks.html#C.IHDR
+  dimensions = png_body[0x10..0x18].unpack('NN')
+  {'x': dimensions[0], 'y': dimensions[1]}
```

Comment:
Note for (way) future reference, may want to eventually run assets through `pngquant` (though those transformations could do with a visual inspection as sometimes it doesn't work with all images and there end up being small artifacts—may be better to just do as periodic batch one-off operations down the road)